pkg/bgpv1/agent/controller.go

```diff
@@ -143,49 +177,73 @@ type ControllerOpt func(*Controller)
 //
 // Cancel the provided CTX to stop the Controller.
 func NewController(ctx context.Context, rtMgr BGPRouterManager, opts ...ControllerOpt) (*Controller, error) {
-	// create the signaler we will use
-	sig := NewSignaler()
-
-	// informers setup
-	clientset := k8s.CiliumClient()
-	factory := externalversions.NewSharedInformerFactory(clientset, 0)
-
-	selfTweakList := informers.WithTweakListOptions(func(lo *metav1.ListOptions) {
-		lo.FieldSelector = "metadata.name=" + nodetypes.GetName()
-	})
-	k8sfactory := informers.NewSharedInformerFactoryWithOptions(k8s.Client(), 0, selfTweakList)
+	var (
+		// signaler used to trigger Controller reconciliation.
+		sig = NewSignaler()
+		// stop channel used to cancel informers.
+		stop = make(chan struct{}, 1)
+		// we'll use to list and watch BGPPeeringPolicies and if IPAM is ClusterPool
+		// CiliumNodes as well.
+		factory = externalversions.NewSharedInformerFactory(k8s.CiliumClient(), 0)
+		// an abtraction over obtaining Node Spec information depending on
+		// IPAM configuration.
+		nodeSpecer nodeSpecer
+	)
 
-	nodeLister := k8sfactory.Core().V1().Nodes().Lister()
-	nodeInformer := k8sfactory.Core().V1().Nodes().Informer()
-	nodeInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{
-		AddFunc:    sig.Event,
-		UpdateFunc: func(_ interface{}, _ interface{}) { sig.Event(struct{}{}) },
-		DeleteFunc: sig.Event,
-	})
+	// setup is dictate by the type of IPAM being used. If Kubernetes IPAM is
+	// being used the BGP Control Plane must watch and retrieve PodCIDR ranges
+	// from the corev1 Kubernetes Node resource.
+	//
+	// if the ClusterPool (both v1 and v2) IPAM mode is in use, the control plane
+	// must watch and retrieve PodCIDR ranges from Cilium's v2 CiliumNode resource.
+	switch option.Config.IPAM {
+	case ipamOption.IPAMClusterPool:
+		fallthrough
+	case ipamOption.IPAMClusterPoolV2:
```

Comment:
tiny nit:
```suggestion
	case ipamOption.IPAMClusterPool, ipamOption.IPAMClusterPoolV2:
```

Reply:
oh nice! forgot about that syntax, I like this more. :+1: